Sources/StreamChat/Database/DTOs/CurrentUserDTO.swift

```diff
@@ -139,35 +139,8 @@ extension NSManagedObjectContext: CurrentUserDatabaseSession {
         }
     }
 
-    private static let currentUserKey = "io.getStream.chat.core.context.current_user_key"
-    private static let removeAllDataToken = "io.getStream.chat.core.context.remove_all_data_token"
-
     var currentUser: CurrentUserDTO? {
-        // we already have cached value in `userInfo` so all setup is complete
-        // so we can just return cached value
-        if let currentUser = userInfo[Self.currentUserKey] as? CurrentUserDTO {
-            return currentUser
-        }
-
-        // we do not have cached value in `userInfo` so we try to load current user from DB
-        if let currentUser = CurrentUserDTO.load(context: self) {
-            // if we have current user we save it to `userInfo` so we do not have to load it again
-            userInfo[Self.currentUserKey] = currentUser
```

Comment:
Why do we remove this caching? Don't think it has an impact on the crash, while it might impact performance.

Reply:
This was also a source of potential crashes. The `userInfo` is not thread-safe so we should not be using it for caching. Besides that, we do not have that observer anymore, so there would be no way to erase this. I think for now this should have a big impact, but if we want to have caching, it needs to be done in higher-level, not in the DTO directly

Reply:
For the current user, it makes sense to have caching - it's frequently being accessed. Yes, please try to keep the caching, on a higher-level is also fine.

Reply:
Moving it to a higher-level actually required a lot of work, so for now I'm leaving it like it was. But now we properly invalidate the cache. Before, there could even be an issue where we did not properly invalidate the current user since we only deleted the cache on logout. But if the current user changes, we also need to invalidate the cache.